ai_chatbots/chatbots.py

```diff
@@ -461,57 +473,135 @@ def get_llm(self, **kwargs) -> BaseChatModel:
+            self.chat_history = json_to_messages(  # noqa: RUF005
+                json_history.get("chat_history", [])
+            ) + [HumanMessage(content=message)]
+            self.intent_history = json_history.get("intent_history", [])
+            self.assessment_history = json_history.get("assessment_history", [])
         else:
             self.chat_history = [HumanMessage(content=message)]
-            self.intent_history = '[]'
-            self.assessment_history = ''
+            self.intent_history = "[]"
+            self.assessment_history = ""
 
         response = ""
 
         try:
             json_output = message_tutor(
-                self.problem, 
-                self.solution, 
-                self.llm, 
-                messages_to_json([HumanMessage(content=message)]), 
-                messages_to_json(self.chat_history), 
-                self.assessment_history, 
-                self.intent_history, 
+                self.problem,
+                self.solution,
+                self.llm,
+                messages_to_json([HumanMessage(content=message)]),
+                messages_to_json(self.chat_history),
+                self.assessment_history,
+                self.intent_history,
                 {"assessor_client": self.llm},
-                tools=tutor_tools
+                tools=tutor_tools,
             )
 
             await create_tutorbot_output(self.thread_id, json_output)
             prossessed = process_StratL_json_output(json_output)
             response = "An error has occurred, please try again"
             for index, msg in enumerate(prossessed[0]):
-                if isinstance(msg, ToolMessage) and msg.name == 'text_student':
-                    response = prossessed[0][index-1].tool_calls[0]['args']['message_to_student']
+                if isinstance(msg, ToolMessage) and msg.name == "text_student":
+                    response = prossessed[0][index - 1].tool_calls[0]["args"][
+                        "message_to_student"
+                    ]
 
             yield response
 
         except Exception:
             yield '<!-- {"error":{"message":"An error occurred, please try again"}} -->'
             log.exception("Error running AI agent")
 
+
+class VideoGPTAgentState(AgentState):
+    """
+    State for the video GPT bot. Passes transcript_asset_id
+    to the associated tool function.
+    """
+
+    transcript_asset_id: Annotated[list[str], add]
+
+
+class VideoGPTBot(BaseChatbot):
+    """Service class for the AI video chat agent"""
+
+    TASK_NAME = "VIDEO_GPT_TASK"
+    JOB_ID = "VIDEO_GPT_JOB"
+
+    INSTRUCTIONS = """You are an assistant helping users answer questions related
+to a video transcript.
+Your job:
+1. Use the available function to gather relevant information about the user's question.
+2. Provide a clear, user-friendly summary of the information retrieved by the tool to
+answer the user's question.
```

Comment:
The issue mentions using this bot to generate both summaries and flash cards, so the instructions might have to include conditional instructions based on the type of desired output (summary vs flashcard vs answer to user question).  But that condition of `"operation": "summary"` vs `"operation": "flashcard"` is not part of this PR, so I'm guessing that will be part of a followup PR?